src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java

```diff
@@ -341,11 +341,21 @@ public static int getFuelValue(@Nonnull ItemStack itemStack)
         int fuelValue = 0;
         for (IFuelHandler handler : fuelHandlers)
         {
-            fuelValue = Math.max(fuelValue, handler.getBurnTime(itemStack));
+            if(handler.matches(itemStack))
+                fuelValue = Math.max(fuelValue, handler.getBurnTime(itemStack));
         }
         return fuelValue;
     }
 
+    public static boolean isFuelHandled(@Nonnull ItemStack itemStack){
+        boolean handled = false;
+        for(IFuelHandler handler : fuelHandlers) {
+            if(handler.matches(itemStack))
+                handled = true;
```

Comment:
You should just return true here, and return false after the for loop. If it has already matched, you can stop searching and avoid the overhead.